src/net35/Radical/Observers/BrokerObserver.cs

```diff
@@ -77,9 +80,9 @@ public MessageBrokerMonitor( IMessageBroker broker )
         /// </summary>
         /// <typeparam name="TMessage">The type of the message.</typeparam>
         /// <returns>This monitor instance.</returns>
-        public MessageBrokerMonitor WaitingFor<TMessage>() where TMessage : class
+        public MessageBrokerMonitor WaitingFor<TMessage>( InvocationModel invocationModel = InvocationModel.Default ) where TMessage : class
```

Comment:
To add more, optional parameters are a hidden nightmare, they work from the compiler perspective as Enums, the default value is copied all around :cry: 

this means that is I have something like:

```
void Foo( int x = 0 )
```

your code can look like:

```
Foo();
```

however your compiled code code looks like

```
Foo(0);
```

Unfortunately the introduction of named and default arguments from the C# perspective is evil :joy: basically they make you leak the internal behavior